docs/email-someone-if-shipping-address-fsa-is-out-of-delivery-area/script.liquid

```diff
@@ -0,0 +1,33 @@
+{% assign available_fsa = options.allowed_fsa__required | split: "," %}
+{% assign order_fsa = order.shipping_address.zip | slice: 0, 3 %}
+
+{% if available_fsa contains order_fsa %}
+	{% assign order_disqualified = false %}
+  {% log %}
+    {{ "The Fsa (" | append: order_fsa | append: ") for order: " | append: order.name | append: " should be in our delivery area." | json }}
+  {% endlog %}
+{% else %}
+	{% assign order_disqualified = true %}
+{% endif %}
+
+{% if event.preview or order_disqualified %}
```

Comment:
Optional (!) sidequest: [switch over to dynamic preview actions](https://docs.usemechanic.com/article/335-preview-actions).

This would look like adding something like this to the very _top_ of this script:

```
{% if event.preview %}
  {% assign order = hash %}
  {% assign order["shipping_address"] = hash %}
  {% assign order["shipping_address"]["name"] = "#1234" %}
  {% assign order["shipping_address"]["zip"] = "XXX123" %}
{% endif %}
```

With that done, you can then remove the `event.preview or` bit from _this_ line. You'll be _able_ to remove that, because the `order` variable (in preview mode _only_) now contains data that you _know_ will trigger the email. (Unless Canada has an XXX FSA. They don't, right?)

I'm making this an optional sidequest; go for it if you're into it, or I'll follow up after we get this merged and get that done for you. :) We consider dynamic preview actions to be a best practice, because they result in _all_ of your task code actually getting _used_ during preview, instead of just saying "oh we're in preview mode let's always render this email action", and risking a logical surprise when a live event comes in.

Reply:
You're right! Canadian FSAs are always in "letter-number-letter" format, so "XXX123" doesn't exist. I added this to the code but I may have misunderstood the sidequest. Thank you so much for taking the time to review this!